[RFC] Doc cleanup (5)

Pyrohh opened the pull request:
I figure smaller, more frequent documentation related PRs is better than larger ones periodically, as the latter could get blocked by something easier, or won't be reviewed. As mentioned in one of the commit messages, this is pretty low hanging fruit.

Commit:
doc: Remove refs to xsmp, xterm_(clipboard|save)

Nvim has no direct integration with X.

Helped-by: Florian Walch <florian@fwalch.com>

Commit:
doc: Remove refs to dos(16|32), win16, and win95

This is primarily low-hanging fruit; there are still many references to
things such as MS-DOS throughout (see os_dos.txt).

cztchoice commented on runtime/doc/os_win32.txt:
```diff
@@ -7,8 +7,8 @@
 
-The Win32 version of Vim works on Windows NT, 95, 98, ME, XP, Vista and
-Windows 7.  There are both console and GUI versions.
+The Win32 version of Vim works on Windows NT, XP, Vista and Windows 7.
```
Can we add windows 8?

Pyrohh replied:
Probably, but I don't want to do it right now since Windows support isn't even merged in yet.

Commit:
README.md: Various

`:h nvim_terminal_emulator.txt` is much more comprehensive than
`:h :terminal`, so link to it instead.

Pyrohh commented:
I added a small commit regarding README.md; that'll be all for now.

Pyrohh commented:
@fmoralesc @fwalch could you take a look? This one doesn't touch the syntax files, so should be more straightforward to review.

fmoralesc commented on runtime/doc/todo.txt:
```diff
@@ -2091,14 +2089,6 @@ MSDOS and Win32:
-
 Win32 console:
 9   When editing a file by its short file name, it should be expanded into its
     long file name, to avoid problems like these: (Mccollister)
```
Does nvim still have shor/long name support? I think I saw it removed somewhere.

justinmk replied:
We still support `%:8`. Not sure if that is what this is about though.

chrisbra replied:
Hi Justin!

On Do, 02 Jul 2015, Justin M. Keyes wrote:

> > @@ -2091,14 +2089,6 @@ MSDOS and Win32:
> >      backslashes. (Ronald Hoellwarth)
> >  
> >  
> > -Windows 95:
> > -8   Editing a file by its short file name and writing it, makes the long file
> > -    name disappear.  Setting 'backupcopy' helps.
> > -    Use FindFirstFile()->cAlternateFileName in fname_case() (George Reilly).
> > -8   Doing wildcard expansion, will match the short filename, but result in the
> > -    long filename (both DJGPP and Win32).
> > -
> > -
> >  Win32 console:
> >  9   When editing a file by its short file name, it should be expanded into its
> >      long file name, to avoid problems like these: (Mccollister)
> 
> We still support `%:8`. Not sure if that is what this is about though.

Are you sure? I believe this was a windows specific so was only 
available in Windows built. And since Windows support was removed, I 
wouldn't bet, that :8 actually does anything currently.

Best,
Christian
-- 
Was die neuen Unwissenden holen müssen:
Frequenzbiegezangen

Pyrohh replied:
I'll leave it for now. Windows support isn't even merged yet, and I don't know much at all about long names and whatnot.

justinmk replied:
I mean, for the Windows build.

Pyrohh replied:
> I mean, for the Windows build.

I must be out of the loop; do we have a working windows build on appveyor? I thought that was just making sure that nvim compiled.

fmoralesc commented on runtime/doc/todo.txt:
```diff
@@ -3158,8 +3140,7 @@ Performance:
     when starting up in a color terminal.
-8   Avoid alloc() for scratch buffer use, esp. in syntax.c.  It's very slow on
-    Win16.
+8   Avoid alloc() for scratch buffer use, esp. in syntax.c.
```
Is this still needed, since neovim has no support for Win16? (Just wondering).

Pyrohh replied:
Nah, I suppose not. Since we don't know that this is an actual issue, I'll remove it.

Pyrohh commented:
Thanks all, re-pushed.

fmoralesc commented on runtime/doc/options.txt:
```diff
@@ -5468,9 +5468,6 @@ A jump table for the options with a short description can be found at |Q_op|.
-	For Dos 32 bits (DJGPP), you can set the $DJSYSFLAGS environment
-	variable to change the way external commands are executed.  See the
-	libc.inf file of DJGPP.
 	Under MS-Windows, when the executable ends in ".com" it must be
```
(Just as as thought for now, since windows support is not yet merged and we can't readily test) Are lines 5474-5476 relevant still? Does it make sense to support [`command.com`](https://en.wikipedia.org/wiki/COMMAND.COM) or [`4dos.com`](https://en.wikipedia.org/wiki/4DOS)?

Pyrohh replied:
Probably not, I removed it and repushed. @equalsraf do you know if we should keep this?

Pyrohh commented:
Rebased and squashed the review commit; I'll merge soon if there's no more comments.

justinmk commented on runtime/doc/options.txt:
```diff
@@ -5461,17 +5461,7 @@ A jump table for the options with a short description can be found at |Q_op|.
-	Under MS-Windows, when the executable ends in ".com" it must be
-	included.  Thus setting the shell to "command.com" or "4dos.com"
-	works, but "command" and "4dos" do not work for all commands (e.g.,
-	filtering).
```
Pretty sure this is still going to be relevant to our Windows port.

justinmk commented:
LGTM
